src/query/sensor/sensor.module.ts

```diff
@@ -39,6 +42,7 @@ export class SensorQueryModule implements OnModuleInit {
 
     const onEvent = (_, event) => {
       this.sensorProcessor.process(event);
```

Comment:
Ik vraag me af of de sensorGateway getriggerd moet worden als er een event uit de eventstore binnen komt of als de processor klaar is met bijwerken? Misschien dat de sensorGateway geïnjecteerd moet worden in de processor en als deze de (mongo)db heeft bijgewerkt, moet de gateway een schop krijgen om een nieuwe stand uit de db te halen en te pushen naar de frontend?

Reply:
Goeie, process() is natuurlijk asynchroon, dit is dus inderdaad iets te optimistisch. Beter om te wachten op bevestiging van Mongo. De emit() kan toegevoegd worden in elke aparte process...() functie, dan hebben we ook keus en welke events wel en niet gedeeld worden. Emitten in process() is cleaner, want maar op 1 plek, maar dan moeten we de Mongo bewerkingen awaiten oid, zodat die flow weer klopt